keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,232 @@
+	// jobs that don't set the label kueue.x-k8s.io/queue-name.
+	// If set to true, then those jobs will be suspended and never started unless
+	// they are assigned a queue and eventually admitted. This also applies to
+	// jobs created before starting the kueue controller.
+	// Defaults to false; therefore, those jobs are not managed and if they are created
+	// unsuspended, they will start immediately.
+	ManageJobsWithoutQueueName bool `json:"manageJobsWithoutQueueName"`
+
+  // ManageAllJobsNamespaceSelector can be used to omit some namespaces from ManageJobsWithoutQueueName
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+
+  ...
+}
+```
+
+### Implementation
+
+The configuration of all Job webhooks are extended to propagate the namespace selector and it is
+used to filter `manageJobsWithoutQueueName` as is currently done in the Pod integration. Specifically,
+jobs without queue names are only managed if both `manageJobsWithoutQueueName` is true and
+the jobs namespace matches the `managedJobsNamespaceSelector`.
+
+Configuration parsing during startup is extended to give a deprecation notice if `podOptions.namespaceSelector` is set.
+Setting both `podOptions.namespaceSelector` and `managedJobsNamespaceSelector` will be flagged as a configuration error.
+
+### Test Plan
+
+[ X ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Unit Tests
+
+Unit tests for configuration parsing will be added.
+
+All unit tests for the current Pod integration namespaceSelector will be extended across all integrations.
+
+#### Integration tests
+
+All integration tests for the current Pod integration namespaceSelector will be extended across all integrations.
+
+## Implementation History
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+### Do nothing
+
+We could do nothing and simply document that despite its historical placement in
+a subfield of the `Configuration` struct, `podOptions.namespaceSelector` also applies
+to the `StatefulSet` and `Deployment` integrations.
+
+### Validating Admission Policies
+
+As previously discussed in [issue 2119](https://github.com/kubernetes-sigs/kueue/issues/2119),
+we could phase out `manageJobsWithoutQueueNames` and instead recommend the
+use of ValidatingAdmissionPolicies to ensure that all Kinds that could be managed by Kueue that
+are created in "user" namespaces have a queue name label (and thus will be managed by Kueue).
+
+However, while exploring this idea further a complication arose. The ValidatingAdmissionPolicy
+must be able to distinguish between top-level resources without queue names (which should be denied)
+and child resources that are owned by Kueue-managed parents (which must be admitted even if they do not have a queue name).
```

Comment:
Yes, I can understand why VAP becomes complicated when there are long chains of CRDs. I think this picture may be even more complex than in this example if we have multiple parent CRDs which can own batch/v1 Job.

Also, introducing `ManagedJobsNamespaceSelector` isn't really adding new complexity assuming we deprecate and drop podOptions.namespaceSelector over time (another comment)

Reply:
I think this is the key point.  For either `manageJobsWithoutQueueName` or the default LocalQueue name being pursued in #3610 to be usable, there _has_ to be a way to control which namespaces they apply to for at least the `Pod`, `StatefulSet` and `Deployment` integrations.   On OpenShift, which tends to use `Job` for system operations the namespace control needs to also cover the `Job` integration as well. 

I can rework this KEP to focus on just the generalization of `integrations.podOptions.namespaceSelector` to `managedJobsNamespaceSelector`.    I think we need it independent of what happens to `manageJobsWithoutQueueName`.